In [1]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [2]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')
df.head()

cost       q       pl      sl      pk      sk      pf      sf
0  0.2130     8.0  6869.47  0.3291  64.945  0.4197  18.000  0.2512
1  3.0427   869.0  8372.96  0.1030  68.227  0.2913  21.067  0.6057
2  9.4059  1412.0  7960.90  0.0891  40.692  0.1567  41.530  0.7542
3  0.7606    65.0  8971.89  0.2802  41.243  0.1282  28.539  0.5916
4  2.2587   295.0  8218.40  0.1772  71.940  0.1623  39.200  0.6606

In [3]:
# специфицируем модель через формулу
output_eq1 = smf.ols(formula='np.log(cost)~np.log(q)+np.log(pl)+np.log(pf)+np.log(pk)', data=df).fit()
# Коэфициенты модели с округление
output_eq1.params.round(3)

Intercept    -7.472
np.log(q)     0.838
np.log(pl)    0.044
np.log(pf)    0.713
np.log(pk)    0.188
dtype: float64

### Спецификация 
$$
log(cost)=-7.472+0.838*log(q)+0.044*log(pl)+0.713*log(pf)+0.188*log(pk)
$$

### Интерпретация 
При увеличении общего выпуска электроэнергии на 1% издержки увеличатся на 0.838% при прочих равных условиях.